pkg/filesystem/subvolume.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subvolume
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	exec "github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type fsStruct struct {
+	Name string
+	data []string
+}
+
+type subVolumeInfo struct {
+	svg string
+	fs  string
+}
+
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace, includeStale string) map[string]subVolumeInfo {
```

Comment:
Why does List() need a return value? They are immediately being printed, and I don't see any other caller of List() that uses the return values.
```suggestion
func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace, includeStale string) {
```

Reply:
this is done for delete case, so that we can use the same function to get stale subvolumes before deleting them.

Reply:
I see now, we need some refactoring for a few reasons. When we delete a subvolume:
1. We don't want to retrieve all possible stale subvolumes across all filesystems. We should be as efficient as possible and only make the queries needed to determine if the subvolume to delete is actually stale. There is no need to evaluate all subvolumes. Let's find a way to factor out the listing and computation of a stale subvolume to be as efficient as possible for deleting a single stale subvolume. 
2. The list results shouldn't be printed.